File: app/src/main/java/com/group15/fitnesstracker/db/SetExercise.java
package com.group15.fitnesstracker.db;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(foreignKeys = @ForeignKey(
                entity = Workout.class,
                parentColumns = "workoutId",
                childColumns = "workoutId")
)
public class SetExercise {
    public SetExercise(String name, String description) {
        this.name = name;
        this.description = description;
    }

    @PrimaryKey
    @NonNull
    public String name;

    public int workoutId;

    public String description;
// @TODO: cant save arrays
//    public Set[] sets;
}

class Set {
    public int reps;
    public double weight;
}

File: app/src/main/java/com/group15/fitnesstracker/db/SetExercise.kt
package com.group15.fitnesstracker.db

import androidx.annotation.NonNull
import androidx.room.*

@Entity
data class SetExercise (
    @PrimaryKey(autoGenerate = true) var exerciseId: Int = 0,
    @NonNull var name: String = "",
    var description: String = "",
    @Ignore var isSelected: Boolean = false
)
